### ISSUE TITLE
Feature: Ability to disable [explicit-module-boundary-types] and [explicit-function-return-type] with RegEx pattern

### ISSUE THREAD

**Issue by rschristian** [typescript-eslint/typescript-eslint#3365]
- [ N/A ] I have tried restarting my IDE and the issue persists.
- [ X ] I have updated to the latest version of the packages.
- [ X ] I have [read the FAQ](https://github.com/typescript-eslint/typescript-eslint/blob/master/docs/getting-started/linting/FAQ.md) and my problem is not listed.

**Repro**

What I'd like to do is something like the following:

```JSON
{
  "rules": {
    "@typescript-eslint/explicit-module-boundary-types": ["warn", { "ignorePattern": "^[A-Z].*" }],
    "@typescript-eslint/explicit-function-return-type": ["warn", { "ignorePattern": "^[A-Z].*" }]
  }
}
```

**Desired Result**

```TS
function Foo() { }   <-- Ignored, allowed
function bar() { }    <-- Warning, no explicit return type
```

I'd like to use RegEx to ignore some pattern, similar to ESLint's [`varsIgnorePatterns`](https://eslint.org/docs/rules/no-unused-vars#varsignorepattern) on the rule `no-unused-vars`.

There's some functions I'd like to ignore with RegEx, as typing them is more trouble than they're ever worth. Adding an `ignorePattern` like shown would be great.

**Comment by bradzacher** [comment]
duplicate of #541

**Comment by rschristian** [comment]
Apologies, I did. GitHub search certainly leaves something to be desired
